File: NCore/World/Block.cs
using global::System.Drawing;

namespace NCore
{
    public class Block
    {
        public Point Position { get; set; }
        public EnumBlockType Type { get; set; }
        public bool Unbreakable { get; set; }
        public bool IsBackground { get; set; }
        public Rectangle Rectangle { get; set; }

        public Block(Point arg0, EnumBlockType arg1, bool arg2, bool arg3)
        {
            Position = arg0;
            Type = arg1;
            Unbreakable = arg2;
            IsBackground = arg3;
            Rectangle = new Rectangle(Position, new Size(32, 32)); 
        }
    }
}

File: NetcraftServer/World/Block.vb
Public Class Block

    Property Position As Point
    Property Type As EnumBlockType
    Property Unbreakable As Boolean
    Property IsBackground As Boolean
    Property Rectangle As Rectangle = New Rectangle(Position, New Size(32, 32))

    Public Sub New(arg0 As Point, arg1 As EnumBlockType, arg2 As Boolean, arg3 As Boolean)
        Position = arg0
        Type = arg1
        Unbreakable = arg2
        IsBackground = arg3
    End Sub

End Class
